### PR TITLE
rootless: add host devices with --privileged

### PR DESCRIPTION
**Pull request by giuseppe**
when --privileged is specified, add all the devices that are usable by
the user.

Closes: https://bugzilla.redhat.com/show_bug.cgi?id=1730773

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@mheon @TomSweeneyRedHat @vrothberg @baude @ashley-cui PTAL

**Comment by rhatdan** [comment]
@giuseppe Your tests is failing...

File: pkg/spec/config_linux.go [hunk 1]
```diff
@@ -118,8 +119,41 @@ func (c *CreateConfig) addPrivilegedDevices(g *generate.Generator) error {
...
+			}
+			if d.Path == "/dev/ptmx" || strings.HasPrefix(d.Path, "/dev/tty") {
+				continue
+			}
+			if _, found := mounts[d.Path]; found {
+				continue
+			}
+			st, err := os.Stat(d.Path)
+			if err != nil {
+				return errors.Wrapf(err, "stat %s", d.Path)
```
Comment (mheon) [inline pkg/spec/config_linux.go]: Could the stat fail on EPERM? Should we just skip if so?

Comment (giuseppe) [reply pkg/spec/config_linux.go]: I thought in that case we still want to skip the test.  It is more about checking if `/dev/kvm` is usable, that if it exists

Comment (mheon) [reply pkg/spec/config_linux.go]: This isn't the test - this is the code for adding devices?

**Comment by TomSweeneyRedHat** [comment]
Other than @mheon's question, LGTM.

File: pkg/spec/config_linux.go (hunk 1 above)
Comment (giuseppe) [reply pkg/spec/config_linux.go]: yes sorry, I've added the code to skip EPERM.  It should not really happen unless `/dev` is not accessible and in that case we wouldn't be able even to use standard devices like `random` or `null`

**Pull request merged**